src/main/java/Prune/Parser.java

```diff
@@ -0,0 +1,255 @@
+            break;
+        case FIND_COMMAND:
+            TaskList matchingTasks = processFinding(inputArray);
+            displayMatchingTasks(matchingTasks);
+            break;
+        default:
+            throw new InvalidCommand(String.format("\tUnknown command: %s\n\tPlease try again", command));
+        }
+
+        // Write data to file for mark, todo, deadline, event, delete commands
+        if (task != null) {
+            this.storage.writeData(this.tasks);
+        }
+    }
+
+    /**
+     * Prints all tasks in tasks list.
+     */
+    public void displayTasks() {
+        if (this.tasks.isEmpty()) {
+            System.out.println("\tHooray! There are no tasks in your list.");
+        } else {
+            System.out.println("\tHere are the tasks in your list:");
+            this.tasks.printTasks();
+        }
+    }
+
+    /**
+     * Prints all matching tasks found.
+     */
+    public void displayMatchingTasks(TaskList matchingTasks) {
+        if (matchingTasks.isEmpty()) {
+            System.out.println("\tThere are no matching tasks found.");
+        } else {
+            System.out.println("\tHere are matching tasks in your list:");
+            matchingTasks.printTasks();
+        }
+    }
+
+    /**
+     * Prints a message after a task has been added.
+     *
+     * @param task Task that has been added.
+     */
+    public void printAddTaskMsg(Task task) {
+        System.out.printf("\tGot it. I've added this task:\n\t\t%s", task);
+        System.out.printf("\n\tNow you have %d tasks in the list.\n", this.tasks.tasksList.size());
+    }
+
+    /**
+     * Prints a message after a task has been deleted.
+     *
+     * @param task Task that has been deleted.
+     */
+    public void printDeleteTaskMsg(Task task) {
+        System.out.printf("\tGot it! I've removed this task:\n\t\t%s", task);
+        System.out.printf("\n\tNow you have %d tasks in the list.\n", this.tasks.tasksList.size());
+    }
+
+    /**
+     * Prints a message after a task has been marked as done or not done.
+     *
+     * @param task Task whose is done status has been changed.
+     */
+    public void printMarkTaskMsg(Task task) {
+        if (task.getIsDone()) {
+            System.out.println("\tNice! I've marked this task as done:");
+        } else {
+            System.out.println("\tOk, I've marked this task as not done yet:");
+        }
+        System.out.printf("\t\t%s\n", task.toString());
+    }
+
+    /**
+     * Processes the saved data when loading from a file.
+     *
+     * @param input Input string containing the task data.
+     */
+    public void processSavedData(String input) {
+        String[] inputArray = input.split(" ", 3);
+        String markStatus = inputArray[0];
+        boolean isDone = markStatus.equals(MARK_DONE_COMMAND);
+        // Form add task input array by removing mark/unmark command
+        String[] newInputArray = Arrays.copyOfRange(inputArray, 1, inputArray.length);
```

Comment:
I see a couple of low-level string manipulation code here - potentially can extract them to improve SLAP